transcode/transcode.go

```diff
@@ -570,47 +597,99 @@ func transcodeSegment(
 	duration := time.Since(start)
 	metrics.Metrics.TranscodeSegmentDurationSec.Observe(duration.Seconds())
 
-	for _, transcodedSegment := range tr.Renditions {
-		renditionIndex := getProfileIndex(transcodeProfiles, transcodedSegment.Name)
-		if renditionIndex == -1 {
-			return fmt.Errorf("failed to find profile with name %q while parsing rendition segment", transcodedSegment.Name)
+	err = processTranscodeResult(segment, transcodeRequest, sourceSegment, tr, encodedProfiles, targetOSURL, transcodedStats, renditionList, segmentChannel)
+	if err != nil {
+		return fmt.Errorf("failed to process transcode result: %w", err)
+	}
+
+	return nil
+}
+
+// withPipedSource is used to duplicate the reading of the `in` reader in case we need a copy of the contents. If
+// `copySource` is false then the `in` reader is returned as is. Otherwise, then a non-nill buffer will be returned and
+// filled after the returned reader is consumed (if present). If no reader is returned (empty transcodeProfiles) the
+// buffer will be already filled with the contents of the `in` reader.
+func withPipedSource(in io.Reader, copySource bool, transcodeProfiles []video.EncodedProfile) (io.Reader, *bytes.Buffer, error) {
+	if !copySource {
+		return in, nil, nil
+	}
+
+	source := new(bytes.Buffer)
+	if len(transcodeProfiles) == 0 {
+		// This is a copy-only job, so skip transcoding with the broadcaster
+		_, err := io.Copy(source, in)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to copy source segment: %s", err)
+		}
+		return nil, source, nil
+	}
+
+	// Otherwise if there are profiles to transcode, then tee the source segment to the buffer
+	return io.TeeReader(in, source), source, nil
```

Comment:
is the second return param (`source`) filled with the io.TeeReader call first when it returns? i.e is source just returned as empty buffer?

Reply:
What do you think of only allowing copy profiles on their own or would that end up making the API too weird and confusing? Fishtank will just use copy on its own right?

Reply:
It is actually part of the requirements that we _always_ want copy-only for the source profile of recordings. Not only when no transcoding is requested. So it also behaves similar to the profiles in livestreams themselves, where you don't need to specify that the source is made available, you only handle what transcoding is gonna be made additionally from that.

Reply:
oh i see so for the regular recording scenario that we have now that will change to a source copy-only plus transcoded renditions?